lib/routes/chikubi/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { DataItem } from '@/types';
+import cache from '@/utils/cache';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+interface ListItem {
+    title: string;
+    link: string;
+}
+
+interface ContentSelectors {
+    title: string;
+    description: string[];
+    date?: string;
+}
+
+const contentTypes: Record<string, ContentSelectors> = {
+    doujin: {
+        title: '.doujin-title',
+        description: ['.doujin-detail', '.section', '.area-buy > a.btn'],
+    },
+    video: {
+        title: '.video-title',
+        description: ['.video-data', '.section', '.lp-samplearea a.btn'],
+    },
+    article: {
+        title: '.article_title',
+        description: ['.article_icatch', '.article_contents'],
+        date: '.date time',
+    },
+};
+
+function getContentType(link: string): keyof typeof contentTypes {
+    const typePatterns = {
+        doujin: ['/cg/', '/comic/', '/voice/'],
+        video: ['/nipple-video/'],
+        article: ['/post-'],
+    };
+
+    for (const [type, patterns] of Object.entries(typePatterns)) {
+        if (patterns.some((pattern) => link.includes(pattern))) {
+            return type as keyof typeof contentTypes;
+        }
+    }
+
+    throw new Error(`Unknown content type for link: ${link}`);
+}
+
+export async function processItems(list: ListItem[]): Promise<DataItem[]> {
+    const items = await Promise.all(
+        list.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const detailResponse = await got(item.link);
+                const $ = load(detailResponse.data);
+
+                const contentType = getContentType(item.link);
+                const selectors = contentTypes[contentType];
+
+                const title = $(selectors.title).text().trim() || item.title;
+                const description = selectors.description
+                    .map((selector) =>
+                        $(selector)
+                            .map((_, el) => $(el).clone().wrap('<div>').parent().html())
+                            .get()
+                            .join('')
+                    )
+                    .join('');
+
+                let pubDate: Date | undefined;
+                if (selectors.date) {
+                    const dateStr = $(selectors.date).attr('datetime');
+                    pubDate = dateStr ? parseDate(dateStr) : undefined;
+                }
```

Comment:
Consider checking the content of `<meta property="article:published_time">` as well